From 7b47a0956ed8b6f8284b6cc8cd8ddaa3c0af3975 Mon Sep 17 00:00:00 2001
From: Brian Koropoff <bkoropoff@gmail.com>
Date: Thu, 10 Mar 2011 20:19:16 +0800
Subject: [SHELL] Port to AIX

- AIX lacks a WCOREDUMP macro.  It's just used to
  append "(core dumped)" to the crash message, so
  #ifdef around it.

- For some reason, the nl program on AIX defaults
  to not printing line numbers ("-b n"), even though
  the spec says it should default to "-b t".
  Explicitly pass "-b a" for good measure in mkbuiltins.

Signed-off-by: Brian Koropoff <bkoropoff@gmail.com>
Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
(cherry picked from commit 8c7562f11d4baa3316bc7b8547cf6497181468c5)

Signed-off-by: Jonathan Nieder <jrnieder@gmail.com>
---
 ChangeLog      |    4 ++++
 src/jobs.c     |    2 ++
 src/mkbuiltins |    2 +-
 3 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index f32dced..4f35c81 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,7 @@
+2011-03-10  Brian Koropoff <bkoropoff@gmail.com>
+
+	* Port to AIX.
+
 2011-03-10  Jilles Tjoelker <jilles@stack.nl>
 
 	* Replace GPL noclobberopen code with the FreeBSD version.
diff --git a/src/jobs.c b/src/jobs.c
index 826a9af..f67116e 100644
--- a/src/jobs.c
+++ b/src/jobs.c
@@ -426,9 +426,11 @@ sprint_status(char *s, int status, int sigonly)
 #endif
 		}
 		col = fmtstr(s, 32, strsignal(st));
+#ifdef WCOREDUMP
 		if (WCOREDUMP(status)) {
 			col += fmtstr(s + col, 16, " (core dumped)");
 		}
+#endif
 	} else if (!sigonly) {
 		if (st)
 			col = fmtstr(s, 16, "Done(%d)", st);
diff --git a/src/mkbuiltins b/src/mkbuiltins
index e38ce4c..99107c2 100644
--- a/src/mkbuiltins
+++ b/src/mkbuiltins
@@ -84,7 +84,7 @@ cat <<\!
  */
 
 !
-sed 's/	-[a-z]*//' $temp2 | nl -v 0 | LC_COLLATE=C sort -u -k 3,3 |
+sed 's/	-[a-z]*//' $temp2 | nl -b a -v 0 | LC_COLLATE=C sort -u -k 3,3 |
 tr abcdefghijklmnopqrstuvwxyz ABCDEFGHIJKLMNOPQRSTUVWXYZ |
 	awk '{	printf "#define %s (builtincmd + %d)\n", $3, $1}'
 printf '\n#define NUMBUILTINS %d\n' $(wc -l < $temp2)
-- 
1.7.4.1

